tests/CMakeLists.txt

```diff
@@ -7,9 +7,9 @@
 # nor does it submit to any jurisdiction.
 
 if( HAVE_TESTS )
-
+set (MPI_OPTS "" CACHE STRING "system depended srun or mpirun options (partition, time, account, ...)")
 if( HAVE_MPI AND MPIEXEC )
-  set( LAUNCH ${MPIEXEC} ${MPIEXEC_NUMPROC_FLAG} 1 )
+  set( LAUNCH ${MPIEXEC} ${MPI_OPTS} ${MPIEXEC_NUMPROC_FLAG} 1 )
```

Comment:
In ecbuild_add_test we cater for something similar but name it MPI_ARGS instead. Perhaps you can use the following snippet instead.
https://github.com/ecmwf/ecbuild/blob/develop/cmake/ecbuild_add_test.cmake#L427-L432